src/client/mod.rs

```diff
@@ -90,31 +95,52 @@ impl GaloyClient {
         Ok(me)
     }
 
-    pub fn request_auth_code(&self, phone: String) -> anyhow::Result<bool> {
-        let input = UserRequestAuthCodeInput { phone };
-
-        let variables = user_request_auth_code::Variables { input };
-
-        let response_body =
-            post_graphql::<UserRequestAuthCode, _>(&self.graphql_client, &self.api, variables)
-                .context("issue fetching response")?;
-
-        let response_data = response_body.data.context("Query failed or is empty")?; // TODO: understand when this can fail here
-
-        let UserRequestAuthCodeUserRequestAuthCode { success, errors } =
-            response_data.user_request_auth_code;
-
-        match success {
-            Some(true) => Ok(true),
-            _ if !errors.is_empty() => {
-                println!("{:?}", errors);
-                bail!("request failed (graphql errors)")
+    pub fn request_phone_code(&self, phone: String, nocaptcha: bool) -> std::io::Result<()> {
+        match nocaptcha {
+            false => {
+                let listener = TcpListener::bind("127.0.0.1:0")?;
+                let port = listener.local_addr().unwrap().port();
+                println!(
+                    "Visit http://127.0.0.1:{}/login and solve the Captcha",
+                    port
+                );
+
+                let rt = tokio::runtime::Builder::new_current_thread()
+                    .enable_all()
+                    .build()?;
+                rt.block_on(server::run(listener, phone, self.api.clone())?)
             }
-            Some(false) => {
-                bail!("request failed (success is false)")
-            }
-            _ => {
-                bail!("request failed (unknown)");
+
+            true => {
+                let input = UserRequestAuthCodeInput { phone };
+
+                let variables = user_request_auth_code::Variables { input };
+                let response_body = post_graphql::<UserRequestAuthCode, _>(
+                    &self.graphql_client,
+                    &self.api,
+                    variables,
+                )
+                .expect("issue fetching response");
+
+                let response_data = response_body.data.expect("Query failed or is empty"); // TODO: understand when this can fail here
+                let UserRequestAuthCodeUserRequestAuthCode { success, errors } =
+                    response_data.user_request_auth_code;
+
+                match success {
```

Comment:
You can have a conversion akin to [this](https://github.com/GaloyMoney/galoy-cli/blob/ac8ac9e7c466f0a4294c21cdaa32126f4c7d6031/src/client/mod.rs#L242), but not pressing for now